src/main/java/edu/cornell/kfs/sys/dataaccess/impl/TablePurgeRecordsDaoOjb.java

```diff
@@ -0,0 +1,139 @@
+package edu.cornell.kfs.sys.dataaccess.impl;
+
+import java.sql.Date;
+import java.util.Collection;
+import java.util.List;
+import java.lang.String;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.ojb.broker.query.Criteria;
+import org.apache.ojb.broker.query.QueryFactory;
+import org.joda.time.DateTime;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.krad.bo.PersistableBusinessObjectBase;
+import org.kuali.kfs.krad.service.BusinessObjectService;
+import org.kuali.kfs.sys.util.KfsDateUtils;
+import org.kuali.rice.core.framework.persistence.ojb.dao.PlatformAwareDaoBaseOjb;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+import edu.cornell.kfs.sys.batch.service.TableLookupCriteriaPurgeService;
+import edu.cornell.kfs.sys.businessobject.TableDetailsForPurge;
+import edu.cornell.kfs.sys.dataaccess.TablePurgeRecordsDao;
+
+public class TablePurgeRecordsDaoOjb extends PlatformAwareDaoBaseOjb implements TablePurgeRecordsDao {
+    private static final Logger LOG = LogManager.getLogger(TablePurgeRecordsDaoOjb.class); 
+    private BusinessObjectService businessObjectService;
+    protected ParameterService parameterService;
+    
+    @Override
+    public void purgeRecords(java.util.Date jobRunDate, List<TableDetailsForPurge> tableDetails) {
+        int daysOldToUse;
+        Date dateForPurge;
+        Criteria lookupCriteria;
+        int defaultDaysOld = retrieveDefaultDaysBeforePurgeParameterValue();
+        for (TableDetailsForPurge details : tableDetails) {
+            daysOldToUse = details.isUseDefaultDaysBeforePurgeParameter() ? defaultDaysOld : retrieveDaysBeforePurgeParameterValue(details.getNameSpaceCode(), details.getComponent(), details.getParameterName());
+            dateForPurge = getPurgeDate(jobRunDate, daysOldToUse);
+            lookupCriteria = requestTablePurgeCriteria(details.getServiceImplForPurgeTableLookupCriteria(), dateForPurge);
+            identifyAndRequestRecordsDeletion(details.getBusinessObjectForRecordsTablePurge(), lookupCriteria);
+        }
+    }
+
+    protected Criteria requestTablePurgeCriteria(String serviceImplClassForPurgeTableLookupCriteria, Date dateForPurge) {
```

Comment:
Would this method name be more clear as buildTablePurgeCriteria?

Reply:
Could serviceImplClassForPurgeTableLookupCriteria be a property injected by spring?

Reply:
Changing method name as suggested.
serviceImplClassForPurgeTableLookupCriteria is being injected by spring. It is the last value for the concurEventNotificationTableDetailsForPurge bean in the cu-spring-sys.xml file